src/PerformanceBenchmarkingInformation.cxx.in

```diff
@@ -0,0 +1,127 @@
+#define MAKE_MAP_ENTRY(KEY, VALUE, DESCRIPTION )                                          \
+m_Map.emplace(                                                                            \
+  std::pair< MapKeyType, InformationValueType>(                                           \
+    MapKeyType{ KEY },                                                                    \
+    InformationValueType{ MapValueType{ VALUE }, MapValueDescriptionType{ DESCRIPTION } } \
+  )                                                                                       \
+)
+
+namespace itk {
+
+SimpleFastMutexLock PerformanceBenchmarkingInformation::m_Mutex;
+PerformanceBenchmarkingInformation::Pointer PerformanceBenchmarkingInformation::m_InformationInstance;
+PerformanceBenchmarkingInformation::MapType PerformanceBenchmarkingInformation::m_Map;
+
+PerformanceBenchmarkingInformation::Pointer
+PerformanceBenchmarkingInformation
+::New()
+{
+  return Self::GetInstance();
+}
+
+PerformanceBenchmarkingInformation::Pointer
+PerformanceBenchmarkingInformation
+::GetInstance()
+{
+  MutexLockHolder<SimpleFastMutexLock> mutexHolder(m_Mutex);
+  if (m_InformationInstance.IsNull())
+  {
+    m_InformationInstance = ObjectFactory<Self>::Create();
+    {
+      new PerformanceBenchmarkingInformation(); //constructor sets m_InformationInstance
+    }
+  }
+  return m_InformationInstance;
+}
+
+const PerformanceBenchmarkingInformation::MapType &
+PerformanceBenchmarkingInformation::GetMap()
+{
+  return PerformanceBenchmarkingInformation::GetInstance()->m_Map;
+}
+
+const PerformanceBenchmarkingInformation::MapValueType
+PerformanceBenchmarkingInformation::GetValue(const MapKeyType &&key)
+{
+  const MapType &localMap = PerformanceBenchmarkingInformation::GetInstance()->GetMap();
+
+  auto it = localMap.find(key);
+  if (it != localMap.end())
+  {
+    return it->second.m_Value;
+  }
+  return {""};
+}
+
+const PerformanceBenchmarkingInformation::MapValueDescriptionType
+PerformanceBenchmarkingInformation::GetDescription(const MapKeyType && key)
+{
+  const MapType &localMap = PerformanceBenchmarkingInformation::GetInstance()->GetMap();
+
+  auto it = localMap.find(key);
+  if (it != localMap.end())
+  {
+    return it->second.m_Description;
+  }
+  return {""};
+}
+
+const std::vector<  PerformanceBenchmarkingInformation::MapKeyType >
+PerformanceBenchmarkingInformation::GetAllKeys()
+{
+  std::vector< PerformanceBenchmarkingInformation::MapKeyType > keyVector;
+  keyVector.reserve(30);
+  for( auto elem : PerformanceBenchmarkingInformation::GetInstance()->m_Map )
+  {
+    keyVector.emplace_back( elem.first );
+  }
+  return keyVector;
+}
+
+PerformanceBenchmarkingInformation
+::PerformanceBenchmarkingInformation()
+{
+  MutexLockHolder<SimpleFastMutexLock> mutexHolder(m_Mutex);
+  m_InformationInstance = this; //threads need this
+  m_InformationInstance->UnRegister(); // Remove extra reference
+
+  @MAPPING_VALUES@
+}
+
+} //end namespace itk
```

Comment:
As per the **KWStyle**, there should be a **new line at the end of the file**. I ignore whether this applied (or should apply) to `*.cxx.in` files as well.